packages/lodestar/src/api/impl/beacon/state/state.ts

```diff
@@ -85,7 +86,7 @@ export class BeaconStateApi implements IBeaconStateApi {
         validatorIndex = undefined;
       }
     }
-    if (!validatorIndex) {
+    if (!validatorIndex && validatorIndex !== 0) {
```

Comment:
`validatorIndex == null`? :joy:

Reply:
What about undefined 😅

Reply:
> What about undefined

`variable == null` with only two `=` will be true for `null` and `undefined`

Reply:
Lol wth thats so confusing. 😓

Reply:
Using Javascript extremely confusing type coercion to our advantage =D https://www.freecodecamp.org/news/js-type-coercion-explained-27ba3d9a2839/

I think 
```ts
if (arg != null) {}
if (arg == null {}
```
Are legit uses so you don't have to check for undefined and null, it's kind of a nullish coalescing operator on a if statement